src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,50 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { filterSlice, Status } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const [querty, setQuerty] = useState('');
+  const timerId = useRef(0);
+
+  const showClearButton = querty.trim().length > 0;
+
+  const onStatusChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterSlice.actions.addStatus(e.target.value as Status));
+  };
+
+  const updateQuerty = (value: string) => {
+    window.clearTimeout(timerId.current);
+    timerId.current = window.setTimeout(() => {
+      dispatch(filterSlice.actions.addQuery(value));
+    }, 2000);
+  };
+
+  const onQuertyChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuerty(e.target.value);
+
+    const normalizedValue = e.target.value.trim();
+
+    if (normalizedValue.length > 0) {
+      updateQuerty(normalizedValue);
+    } else {
+      dispatch(filterSlice.actions.removeQuery());
+    }
+  };
+
+  const onQuertyDelete = () => {
```

Comment:
Why do you always start the name of the function from `on` and only `update` function you write without it

And probably you make some typo mistake `Querty` to `query`